### PR TITLE
fix: check for inaccurate filesystem (`mtime`)

### PR DESCRIPTION
**Pull request by sokra**
fix an issue, which caused incorrect timestamps with inaccurate filesystems or fast changes.

### CODE REVIEW

**Comment by sokra** [comment]
@abirmingham Could you test this fix?

File: src/index.js [hunk 1]
```diff
@@ -14,15 +14,21 @@ function loader(...args) {
...
   const dependencies = this.getDependencies().concat(this.loaders.map(l => l.path));
   const contextDependencies = this.getContextDependencies();
+  let allowCache = true;
   const toDepDetails = (dep, mapCallback) => {
     fs.stat(dep, (err, stats) => {
       if (err) {
         mapCallback(err);
         return;
       }
+      const mtimeValue = stats.mtime.getTime();
```
Comment (michael-ciniawsky) [inline src/index.js]: `mtimeValue` => `mtime`

File: src/index.js [hunk 2]
```diff
@@ -14,15 +14,21 @@ function loader(...args) {
...
+      if (mtimeValue / 1000 >= Math.floor(data.startTime / 1000)) {
+        // Don't trust mtime. File was changed while compiling or it could be an inaccurate filesystem.
+        allowCache = false;
+      }
       mapCallback(null, {
         path: dep,
-        mtime: stats.mtime.getTime(),
+        mtime: mtimeValue,
```
Comment (michael-ciniawsky) [inline src/index.js]:
```diff
- mtime: mtimeValue
+ mtime
```

**Comment by abirmingham** [comment]
@sokra works for me! Thanks for looking into this.

**Comment by michael-ciniawsky** [comment]
@sokra Could you rebase, so the CI doesn't fail on node 4? Or merge it :), but I atm can't due to required checks

**Pull request merged**